src/wallet/sqlite.cpp

```diff
@@ -259,22 +259,117 @@ void SQLiteBatch::Close()
 
 bool SQLiteBatch::ReadKey(CDataStream&& key, CDataStream& value)
 {
-    return false;
+    if (!m_database.m_db) return false;
+    assert(m_read_stmt);
+
+    // Bind: leftmost parameter in statement is index 1
+    int res = sqlite3_bind_blob(m_read_stmt, 1, key.data(), key.size(), SQLITE_STATIC);
+    if (res != SQLITE_OK) {
+        LogPrintf("SQLiteBatch::ReadKey: Unable to bind read statement :%s\n", sqlite3_errstr(res));
+        sqlite3_clear_bindings(m_read_stmt);
+        sqlite3_reset(m_read_stmt);
+        return false;
+    }
+    res = sqlite3_step(m_read_stmt);
+    if (res != SQLITE_ROW) {
+        LogPrintf("SQLiteBatch::ReadKey: Unable to execute read statement :%s\n", sqlite3_errstr(res));
+        sqlite3_clear_bindings(m_read_stmt);
+        sqlite3_reset(m_read_stmt);
+        return false;
+    }
+    // Leftmost column in result is index 0
+    const char* data = (const char*)sqlite3_column_blob(m_read_stmt, 0);
```

Comment:
14b81e521c310cb616b7920b5625b704ef1e1899
```suggestion
    const char* data = reinterpret_cast<const char*>(sqlite3_column_blob(m_read_stmt, 0));
```

Reply:
Done